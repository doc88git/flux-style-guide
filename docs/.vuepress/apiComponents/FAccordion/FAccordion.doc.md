### Props

| Prop name | Description     | Type   | Values   | Default |
| --------- | --------------- | ------ | -------- | ------- |
| title     | Accordion title | string | `string` |         |

### Slots

| Name    | Description | Bindings |
| ------- | ----------- | -------- |
| default |             |          |
