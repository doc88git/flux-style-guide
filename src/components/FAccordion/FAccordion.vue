<template>
  <div class="BasicAccordion">
    <div class="BasicAccordion__header">
      <p class="BasicAccordion__header__p">
        {{ title }}
      </p>
      <div class="BasicAccordion__header__info">
        <slot v-if="hideContent" name="info" />
      </div>
      <f-icon
        :class="iconClasses"
        lib="flux"
        name="chevron-up"
        type="outlined"
        size="lg"
        color="black"
        clickable
        @click.native="toggleContent"
      />
    </div>

    <div :class="contentClasses">
      <div class="BasicAccordion__content__inner">
        <slot />
      </div>
    </div>
  </div>
</template>

<script>
import FIcon from '../FIcon/FIcon'

export default {
  name: 'f-accordion',

  components: { FIcon },

  props: {
    /**
     * Accordion title
     * @values string
     * @required true
     */
    title: {
      type: String,
      required: true
    }
  },

  data: () => ({ hideContent: false }),

  computed: {
    contentClasses() {
      return [
        'BasicAccordion__content',
        {
          'BasicAccordion__content--hidden': this.hideContent
        }
      ]
    },

    iconClasses() {
      return [
        'BasicAccordion__header__icon',
        {
          'BasicAccordion__header__icon--rotate': this.hideContent
        }
      ]
    }
  },

  methods: {
    toggleContent() {
      this.hideContent = !this.hideContent
    }
  }
}
</script>

<style lang="scss" scoped>
.BasicAccordion {
  display: flex;
  flex-direction: column;

  background: #fff;
  border-radius: 5px;
  box-shadow: 0px 0px 16px #0000001f;

  &__header {
    display: flex;
    height: 60px;
    padding: 0 35px 0 20px;

    &__info {
      display: flex;
      justify-content: center;
      align-items: center;
      margin-right: 20px;
    }

    &__p {
      display: flex;
      align-items: center;

      flex-grow: 1;
      vertical-align: middle;
      font-size: 16px;
      font-weight: bold;
      color: #666666;
    }

    &__icon {
      display: flex;
      align-items: center;
      transition: transform 300ms;

      &--rotate {
        transform: rotate(180deg);
      }
    }
  }

  &__content {
    max-height: 900px;
    overflow: hidden;
    transition: max-height 300ms ease-in, overflow 0ms ease 300ms;

    &--hidden {
      max-height: 0px;
    }

    &__inner {
      padding: 0 20px 20px 20px;
    }
  }
}
</style>
