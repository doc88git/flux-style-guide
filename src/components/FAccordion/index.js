import FAccordion from './FAccordion'

export { FAccordion }