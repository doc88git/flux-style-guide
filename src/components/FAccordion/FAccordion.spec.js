import { shallowMount } from '@vue/test-utils'
import FAccordion from './FAccordion'

const WRAPPER_PROPS = { title: 'Test' }

describe('FAccordion tests', () => {
  let WRAPPER

  beforeEach(
    () => (WRAPPER = shallowMount(FAccordion, { propsData: WRAPPER_PROPS }))
  )

  test('Componente existe', () => {
    expect(WRAPPER.exists()).toBe(true)
  })

  test('Componente não alterado', () => {
    expect(WRAPPER.html()).toMatchSnapshot()
  })

  describe('Teste de computeds', () => {
    describe('Computed contentClasses', () => {
      test('Computed contentClasses quando hideContent: true', () => {
        WRAPPER.setData({ hideContent: true })
        const expectData = [
          'BasicAccordion__content',
          {
            'BasicAccordion__content--hidden': true
          }
        ]

        expect(WRAPPER.vm.contentClasses).toStrictEqual(expectData)
      })

      test('Computed contentClasses quando hideContent: false', () => {
        WRAPPER.setData({ hideContent: false })
        const expectData = [
          'BasicAccordion__content',
          {
            'BasicAccordion__content--hidden': false
          }
        ]

        expect(WRAPPER.vm.contentClasses).toStrictEqual(expectData)
      })
    })

    describe('Computed iconClasses', () => {
      test('Computed iconClasses quando hideContent: true', () => {
        WRAPPER.setData({ hideContent: true })
        const expectData = [
          'BasicAccordion__header__icon',
          {
            'BasicAccordion__header__icon--rotate': true
          }
        ]

        expect(WRAPPER.vm.iconClasses).toStrictEqual(expectData)
      })

      test('Computed iconClasses quando hideContent: false', () => {
        WRAPPER.setData({ hideContent: false })
        const expectData = [
          'BasicAccordion__header__icon',
          {
            'BasicAccordion__header__icon--rotate': false
          }
        ]

        expect(WRAPPER.vm.iconClasses).toStrictEqual(expectData)
      })
    })
  })

  describe('Teste de métodos', () => {
    test('Método toggleContent', () => {
      expect(WRAPPER.vm.hideContent).toBe(false)
      WRAPPER.vm.toggleContent()
      expect(WRAPPER.vm.hideContent).toBe(true)
    })
  })
})
