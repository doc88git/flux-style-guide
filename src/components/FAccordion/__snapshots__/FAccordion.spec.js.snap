// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAccordion tests Componente n√£o alterado 1`] = `
"<div class=\\"BasicAccordion\\">
  <div class=\\"BasicAccordion__header\\">
    <p class=\\"BasicAccordion__header__p\\">
      Test
    </p>
    <div class=\\"BasicAccordion__header__info\\">
      <!---->
    </div>
    <f-icon-stub color=\\"black\\" lib=\\"flux\\" size=\\"lg\\" name=\\"chevron-up\\" clickable=\\"true\\" type=\\"outlined\\" class=\\"BasicAccordion__header__icon\\"></f-icon-stub>
  </div>
  <div class=\\"BasicAccordion__content\\">
    <div class=\\"BasicAccordion__content__inner\\"></div>
  </div>
</div>"
`;
